### PR TITLE
Adding types to the symbol table - consistent export type sample rule implemented

### PR DESCRIPTION
**Pull request by EliLichtblau**
Fixes 11 TS conformance tests and 18 babel tests.

Adds types to the symbol table functionally conformant to TS behavior but symbol flags implemented slightly differently.

Symbol redeclaration check is also not entirely conformant but fixing this seems like a separate PR.

For testing purposes - consistent-export-type was also implemented (WIP).

Major TODOs:
* Fully implement consistent type exports - (good enough at this point imo)
* Computed property names of enums is currently unimplemented - (This seems out of scope to properly implement actually)
* Tests/Documentation - (If you want me to write tests let me know where but I think conformance tests are good for now?)

### CODE REVIEW

**Comment by EliLichtblau** [comment]
Module merging is actually deeply confusing - there is a small bug right now that 
```ts
module F {}
function F() {}
```
is an error in TS but
```ts
function F() {}
module F {}
```
is valid code - I don't even want to begin to think about how this works across files and I'd be happy to merge this without this fully working.

File: crates/oxc_semantic/src/symbol.rs [hunk 1]
```diff
@@ -54,6 +54,10 @@ impl SymbolTable {
         self.flags[symbol_id]
     }
 
+    pub fn set_flag(&mut self, symbol_id: SymbolId, includes: SymbolFlags) {
+        self.flags[symbol_id] |= includes;
```
Comment (Boshen) [inline crates/oxc_semantic/src/symbol.rs:58]: one last nitpick: this is unioning the flag, which means the function name and documentation should indicate this.

File: crates/oxc_ast/src/ast/ts.rs [hunk 2]
```diff
@@ -17,11 +17,20 @@ pub struct TSEnumDeclaration<'a> {
...
+/// Enum Body
+///
+/// A scope must be created on the enum body so this abstraction exists
+#[derive(Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize), serde(tag = "type"))]
+pub struct TSEnumBody<'a> {
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    pub span: Span,
+    pub members: Vec<'a, TSEnumMember<'a>>,
+}
```
Comment (Boshen) [inline crates/oxc_ast/src/ast/ts.rs:33]: nitpick: a newline here would be nice.

**Review by Boshen — COMMENTED** [review]
I both love and hate how TypeScript logic is getting merged into semantic builder, which makes the code much hard to follow 😢 

But overall this is good, we are passing more tests!

File: crates/oxc_semantic/src/symbol.rs (hunk 1 above)
Comment (EliLichtblau) [reply crates/oxc_semantic/src/symbol.rs:58]: Good point! Thanks!

**Comment by EliLichtblau** [comment]
And this is only bind!:/ - the TS checker - even if you ignore all the options that modulate type checking is wildly complex - the ways in which just choosing CommonJS vs commonjs change type checking i dont understand - but tighter integration with ezno (or writing a different checker) and trying to get typechecking lint rules working would be really cool.

**Review by Boshen — APPROVED** [review]
